cmd/up.go

```diff
@@ -241,23 +245,27 @@ func newUpCmd() *cobra.Command {
 		// - attempt `destroy` on any update errors.
 		// - show template.Quickstart?
 
-		changes, err := s.Update(commandContext(), backend.UpdateOperation{
+		changes, res := s.Update(commandContext(), backend.UpdateOperation{
 			Proj:   proj,
 			Root:   root,
 			M:      m,
 			Opts:   opts,
 			Scopes: cancellationScopes,
 		})
-		switch {
-		case err == context.Canceled:
+
+		if res != nil && res.Error() == context.Canceled {
 			return errors.New("update cancelled")
-		case err != nil:
-			return PrintEngineError(err)
-		case expectNop && changes != nil && changes.HasChanges():
+		}
+
+		if res != nil {
+			return PrintEngineResult(res)
+		}
+
+		if expectNop && changes != nil && changes.HasChanges() {
 			return errors.New("error: no changes were expected but changes occurred")
-		default:
-			return nil
 		}
+
+		return nil
```

Comment:
lots of redundancy in this err/res handling sections.  I'm ok with making a helper for all of these.  but the diff is acutally easier to read in this form first.